File: tests/files/language_test_files/file.go
// File.go

package main

import (
	"fmt"
)

func testcount(x int) int {
	if x == 11 {
		return 0
	}
	fmt.Println(x)
	return testcount(x + 1)
}

func main() {
	// Lambda functions assigned to variables
	add := func(a, b int) int {
		return a + b
	}

	multiply := func(x, y int) int {
		return x * y
	}

	square := func(num int) int {
		return num * num
	}

	// Anonymous lambda function
	func(x int) {
		fmt.Println("Anonymous lambda function with argument:", x)
	}(5)

	fmt.Println("Addition result:", add(3, 5))
	fmt.Println("Multiplication result:", multiply(4, 6))
	fmt.Println("Square result:", square(9))
}

File: tests/files/language_test_files/file.cs
using System;

class Program
{
    // Regular method
    static int Add(int a, int b)
    {
        return a + b;
    }

    // Lambda function assigned to a variable
    static Func<int, int, int> Multiply = (x, y) => x * y;

    // Anonymous method
    static Func<int, int> Square = delegate (int num)
    {
        return num * num;
    };

    static void Main()
    {
        // Usage of each function
        Console.WriteLine("Addition result: " + Add(3, 5));
        Console.WriteLine("Multiplication result: " + Multiply(4, 6));
        Console.WriteLine("Square result: " + Square(9));
    }
}
